### ISSUE TITLE
Scale up fails if a restore was performed in the past

### ISSUE THREAD

**Issue by adejanovski** [k8ssandra/k8ssandra#616]
## Bug Report

**Describe the bug**
Scaling up the Cassandra nodes after having restored a backup will make the new pod loop crash as it tries to perform the restore as well.

Even after having deleted the cassandrabackup/cassandrarestore CRDs, it will still start the init container with a restore key and a backup name for the medusa-restore init container:

```
  medusa-restore:
    Container ID:   containerd://8e2d71cdeb931b43568bd2bd714f700b3c8f397cab3f3f713720a9518d38bd65
    Image:          docker.io/jsanda/medusa:grpc-sigterm-13d70bb2
    Image ID:       docker.io/jsanda/medusa@sha256:2095ea34de678178ff0798b8045390947e1963d6dd376b2669f255d74c822786
    Port:           <none>
    Host Port:      <none>
    State:          Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Thu, 01 Apr 2021 11:14:03 +0200
      Finished:     Thu, 01 Apr 2021 11:14:04 +0200
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Thu, 01 Apr 2021 11:13:45 +0200
      Finished:     Thu, 01 Apr 2021 11:13:47 +0200
    Ready:          False
    Restart Count:  2
    Environment:
      MEDUSA_MODE:   RESTORE
      CQL_USERNAME:  <set to the key 'username' in secret 'k8ssandra-medusa'>  Optional: false
      CQL_PASSWORD:  <set to the key 'password' in secret 'k8ssandra-medusa'>  Optional: false
      BACKUP_NAME:   backup1
      RESTORE_KEY:   4c37944c-82bd-4342-a016-2b2544884afe
```

**To Reproduce**
1. Create a cluster with a single Cassandra node
2. Create a backup
3. Restore the backup
4. After the restore is finished, scale up to 2 Cassandra nodes
5. The newly created pod will go into a crash loop and will fail to initialize

**Expected behavior**
Once a restore is finished, it should not be scheduled on nodes that were not part of the initial backup.
Scaling up should be possible without a restore being attempted.

**Environment (please complete the following information):**

* Helm charts version info

`1.1.0-SNAPSHOT` 

* Kubernetes cluster kind:

kind

**Comment by adejanovski** [comment]
Duplicate of #461
